helix-dap/src/types.rs

```diff
@@ -992,6 +1022,29 @@ pub mod events {
     }
 }
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Serialize)]
+pub enum ConnectionType {
```

Comment:
Instead of manually implementing Deserialize this it should work to `#[derive(Deserialize)]` along with `#[serde(rename_all = "lowercase")]`:

```suggestion
#[derive(Clone, Copy, Debug, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum ConnectionType {
```